fdbclient/include/fdbclient/BulkLoading.h

```diff
@@ -541,64 +539,177 @@ enum class BulkLoadPhase : uint8_t {
 	Error = 6, // Updated by DD when this task has unretriable error
 };
 
+struct BulkLoadManifestSet {
+public:
+	constexpr static FileIdentifier file_identifier = 1384493;
+
+	BulkLoadManifestSet() = default;
+
+	BulkLoadManifestSet(int inputMaxCount) { maxCount = inputMaxCount; }
+
+	bool isValid() const {
+		if (maxCount == 0) {
+			return false;
+		}
+		if (manifests.empty()) {
+			return false;
+		}
+		if (manifests.size() > maxCount) {
+			return false;
+		}
+		if (!minBeginKey.present()) {
+			return false;
+		}
+		if (!maxEndKey.present()) {
+			return false;
+		}
+		for (const auto& manifest : manifests) {
+			if (!manifest.isValid()) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	// Return true if succeed
+	bool addManifest(const BulkLoadManifest& manifest) {
```

Comment:
The manifests in a BulkLoadManifestSet do not need to be in sequence? They do need to belong to the ranges being served by a SS? Or is this not required?  A SS can work on tasks whose keyranges it is not serving?

Reply:
Maybe the ranges need to be in sequence since you take the min and max of the BulkLoadManifestSet... If so, do we need a check in here in addManifest that the added manifest follows the one previously added?

Reply:
We do not need to specify the order of manifests here which is error prone. The SS will re-scan the manifests to build up the manifest key range map before loading it. See `tryGetRangeForBulkLoad` for details.

The only important thing here is that after a task completes all addManifest()'s, the key range is decided which is the min begin key and the max end key among all manifests. So, we should be make sure different tasks do not have overlapping ranges. This is guaranteed by `scheduleBulkLoadJob`

Reply:
We can add manifest a-c and e-g without a manifest that covers d to e? That is ok? d-e could be part of another task set?

Does the SS have to be serving the range it downloads?

Reply:
We do not allow this case. When a task addManifest, the `scheduleBulkLoadJob` checks this case does not happen by asserts. However, the addManifest method itself does not check this. I will add comments in the next PR.